src/transformation/visitors/switch.ts

```diff
@@ -18,30 +18,50 @@ const containsBreakOrReturn = (nodes: Iterable<ts.Node>): boolean => {
 };
 
 const coalesceCondition = (
-    condition: lua.Expression | undefined,
-    switchVariable: lua.Identifier,
-    expression: ts.Expression,
-    context: TransformationContext
-): lua.Expression => {
+    condition: ts.Expression | undefined,
+    switchVariable: ts.Identifier,
+    expression: ts.Expression
+): ts.Expression => {
+    const comparison = ts.factory.createBinaryExpression(
```

Comment:
This looks nice and simple, but don't all these ts.factory.create* methods interfere with source maps and potentially type checking that needs tree traversal?

Reply:
Good point. Wasn't fully considering that. I've reverted to using lua expressions here and manually handled preceding statements.